package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -161,6 +162,27 @@ def add_output(self, modular_pipeline_id: str, output_node: GraphNode):
         else:
             self.tree[modular_pipeline_id].external_outputs.add(output_node.id)
 
+    def add_tags(self, modular_pipeline_id: str, node_tags: set):

```

Comment:
I was going through the modular pipelines code and found that we update the modular_pipeline `pipelines` set with `node.pipelines`. May be we can update the tags at that point too. 

Like at line 239 in this file inside `extract_from_node` function having something like - `modular_pipeline.tags.update(node.tags)`. In that case we do not need this function and also do not need a call at line 203 in `managers.py`. What do you think ? @rashidakanchwala

Reply:
That's how I believe @jitu5 approached it previously. 
He had some code in the `extract_from_node` function, which I found very confusing because the function was originally defined to extract namespaces, but now it's getting tags. 
I think it's important to have clean functions that handle one task each, rather than one function doing too many things.

I'm also personally confused by the `extract_from_node` function being called three times here - [link to GitHub code](https://github.com/kedro-org/kedro-viz/blob/main/package/kedro_viz/data_access/managers.py#L156-L202).
This is the exactly code refactor/simplification that Ivan mentioned to us the other day.